shared/config/config.go

```diff
@@ -0,0 +1,109 @@
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+)
+
+// TODO(olshansky): Should this be an interface so it can be mocked and injected?
+type Config struct {
+	RootDir    string `json:"root_dir"`
+	PrivateKey string `json:"private_key"`
+	Genesis    string `json:"genesis"`
+
+	PREP2P      *PREP2PConfig      `json:"prep2p"` // TODO(derrandz): delete this once P2P is ready.
+	P2P         *P2PConfig         `json:"p2p"`
+	Consensus   *ConsensusConfig   `json:"consensus"`
+	Persistence *PersistenceConfig `json:"persistence"`
+	Utility     *UtilityConfig     `json:"utility"`
+}
+
+// TODO(derrandz): delete this once P2P is ready.
+type PREP2PConfig struct {
+	ConsensusPort uint32 `json:"consensus_port"`
+	DebugPort     uint32 `json:"debug_port"`
+}
+
+type P2PConfig struct {
+	Protocol   string   `json:"protocol"`
+	Address    string   `json:"address"`
+	ExternalIp string   `json:"external_ip"`
+	Peers      []string `json:"peers"`
+}
+
+type ConsensusConfig struct {
+	// TODO(olshansky): This should be assigned dynamically by the consensus module through sorting and validation.
+	NodeId uint32 `json:"node_id"`
+}
+
+type PersistenceConfig struct {
+	DataDir string `json:"datadir"`
+}
+
+type UtilityConfig struct {
+}
+
+func LoadConfig(file string) (c *Config) {
```

Comment:
Good point,  but I think this discussion exceeds the scope of this PR and might be better suited for an issue related to "configuration & parameters" rather than the "app-specific skeleton bus".